File: server/beta-server/src/main/java/ru/fullrest/mfr/test_server/model/entity/TelegramUser.java
package ru.fullrest.mfr.test_server.model.entity;

import lombok.Data;

import javax.persistence.*;

@Data
@Entity
@Table(name = "TelegramUser")
public class TelegramUser {

    @Id
    @Column(name = "UserId")
    private int id;

    @Column(name = "Username")
    private String username;

    @Column(name = "Role")
    @Enumerated(EnumType.STRING)
    private UserRole role;
}

File: server/src/main/kotlin/com/lezenford/mfr/server/model/entity/TelegramUser.kt
package com.lezenford.mfr.server.model.entity

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.EnumType
import javax.persistence.Enumerated
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "TelegramUser")
class TelegramUser(
    @Id
    @Column(name = "Id")
    override val id: Long,

    @Column(name = "Username")
    var username: String,

    @Column(name = "Role")
    @Enumerated(EnumType.STRING)
    var role: Role
) : Base<Long>() {
    enum class Role {
        USER, ADMIN
    }
}

